node/src/reactor/joiner.rs

```diff
@@ -509,52 +488,18 @@ impl reactor::Reactor for Reactor {
             DeployGetter::new(effect_builder),
         )?;
 
-        let block_validator = BlockValidator::new(Arc::clone(chainspec_loader.chainspec()));
-
         let deploy_fetcher = Fetcher::new("deploy", config.fetcher, registry)?;
-
         let block_by_height_fetcher = Fetcher::new("block_by_height", config.fetcher, registry)?;
-
-        let block_header_and_finality_signatures_by_height_fetcher: Fetcher<
-            BlockHeaderWithMetadata,
-        > = Fetcher::new(
-            "block_header_and_finality_signatures_by_height",
-            config.fetcher,
-            registry,
-        )?;
-
+        let block_fetcher = Fetcher::new("block", config.fetcher, registry)?;
+        let trie_fetcher = Fetcher::new("trie", config.fetcher, registry)?;
+        let block_header_and_finality_signatures_by_height_fetcher =
+            Fetcher::new("block_header_by_height", config.fetcher, registry)?;
         let block_header_by_hash_fetcher: Fetcher<BlockHeader> =
```

Comment:
Should this just be `block_header_fetcher` then, since for `block_fetcher` it's also implicit?

Reply:
@afck I can try to resolve this

Reply:
I opted to make `block_fetcher` be `block_by_hash_fetcher` instead (since people like me might be confused about how you querying for them exactly).

Fixed in 95710d4a6